src/main/java/manageezpz/logic/parser/FindEmployeeCommandParser.java

```diff
@@ -0,0 +1,109 @@
+package manageezpz.logic.parser;
+
+import static manageezpz.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static manageezpz.logic.parser.CliSyntax.PREFIX_NAME;
+import static manageezpz.logic.parser.CliSyntax.PREFIX_PHONE;
+
+import java.util.Arrays;
+import java.util.List;
+
+import manageezpz.commons.core.Messages;
+import manageezpz.logic.commands.FindEmployeeCommand;
+import manageezpz.logic.parser.exceptions.ParseException;
+import manageezpz.model.person.Email;
+import manageezpz.model.person.Name;
+import manageezpz.model.person.PersonMultiplePredicate;
+import manageezpz.model.person.Phone;
+
+/**
+ * Checks if the options are valid for finding employees.
+ */
+public class FindEmployeeCommandParser implements Parser<FindEmployeeCommand> {
+    private static final Prefix[] PERSON_PROPERTIES = {PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL};
+
+    private String errorMessage = "";
+    private boolean hasError = false;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public FindEmployeeCommand parse(String userInput) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(userInput, PERSON_PROPERTIES);
+
+        List<String> names = getPersonName(argMultimap);
+        String phone = getPersonPhone(argMultimap);
+        String email = getPersonEmail(argMultimap);
+
+        if (hasError) {
+            String finalMessage = errorMessage + FindEmployeeCommand.MESSAGE_USAGE;
+            String displayedMessage = String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, finalMessage);
+            throw new ParseException(displayedMessage);
+        } else {
+            PersonMultiplePredicate predicate = new PersonMultiplePredicate(names, phone, email);
+            return new FindEmployeeCommand(predicate);
+        }
+    }
+
+    private List<String> getPersonName(ArgumentMultimap argMultimap) {
+        List<String> names = null;
+        if (argMultimap.isPrefixExist(PREFIX_NAME)) {
+            String nameArgumentString = argMultimap.getValue(PREFIX_NAME).get().trim();
+            String[] nameArguments = nameArgumentString.split("\\s+");
+            names = checkIfAllNamesValid(nameArguments);
+        }
+        return names;
+    }
+
+    private List<String> checkIfAllNamesValid(String[] nameArguments) {
+        List<String> names = null;
+        boolean isValid = Arrays.stream(nameArguments).allMatch(name -> Name.isValidName(name));
+        if (!isValid) {
+            addErrorMessage(FindEmployeeCommand.INVALID_NAME);
+        } else {
+            names = Arrays.asList(nameArguments);
+        }
+        return names;
+    }
+
+    private String getPersonPhone(ArgumentMultimap argMultimap) {
+        String phone = null;
```

Comment:
Perhaps better to create String phone as an empty String
i.e. String phone = ""